### PR TITLE
Upgrade Jackson to 2.16.2, embulk-util-config to 0.5.0, and embulk-util-json to 0.4.0

### PR DESCRIPTION
**Pull request by dmikurube**

### CODE REVIEW

File: embulk-guess-csv/gradle.lockfile [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
 # Manual edits can break the build and are not advised.
 # This file is expected to be part of source control.
-com.fasterxml.jackson.core:jackson-annotations:2.6.7=compileClasspath,runtimeClasspath
-com.fasterxml.jackson.core:jackson-core:2.6.7=compileClasspath,runtimeClasspath
-com.fasterxml.jackson.core:jackson-databind:2.6.7.5=compileClasspath,runtimeClasspath
-com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.6.7=compileClasspath,runtimeClasspath
-javax.validation:validation-api:1.1.0.Final=compileClasspath,runtimeClasspath
+com.fasterxml.jackson.core:jackson-annotations:2.15.4=compileClasspath
+com.fasterxml.jackson.core:jackson-annotations:2.16.2=runtimeClasspath
+com.fasterxml.jackson.core:jackson-core:2.15.4=compileClasspath
```
Comment (hiroyuki-sato) [inline embulk-guess-csv/gradle.lockfile:6]: Nit. Could you tell me why `compileClassPath` and `runtimeClasspath` use different jackson version?

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍 (I left one minor comment)

File: embulk-guess-csv/gradle.lockfile (hunk 1 above)
Comment (dmikurube) [reply embulk-guess-csv/gradle.lockfile:6]: Ah, good finding! Let me check...

Comment (dmikurube) [reply embulk-guess-csv/gradle.lockfile]: The guess plugin should have had Jackson BOM, but it didn't have. Fixed it in 7e329b0bbcd8e1f9a147a7cc18722d50005cc043

Thanks for your good catch.

**Comment by hiroyuki-sato** [comment]
LGTM👍

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**